coreutils_core/src/input.rs

```diff
@@ -0,0 +1,97 @@
+//! The Input module handles getting user input from the command line
+
+use std::{io, io::prelude::*};
+
+/// The Input struct handles issuing messages and getting responses from the user.
+/// # Example
+/// To ask the user whether they want to make a change or not, and validate their response
+/// into a `bool`, the `Input` struct can be used like so:
+/// ```rust,ignore
+/// let answer: bool = Input::new()
+///     .with_msg("Do you want to make this change?")
+///     .with_err_msg("Error! Failure to read!")
+///     .is_affirmative();
+///
+/// assert_eq!(answer, true);
+/// ```
+///
+/// One could also get the response directly from the user like so:
+/// ```rust,ignore
+/// let answer: String = Input::new()
+///     .with_msg("Do you want to make this change?")
+///     .with_err_msg("Error! Failure to read!")
+///     .get();
+///
+/// assert_eq!(answer, String::from("Yes, I do"));
+/// ```
+#[derive(Debug, Clone)]
+pub struct Input<'a> {
+    msg:     Option<&'a str>,
+    err_msg: Option<&'a str>,
+}
+
+impl<'a> Default for Input<'a> {
+    fn default() -> Input<'a> { Input { msg: None, err_msg: None } }
+}
```

Comment:
Option always defaults to `None`, since both default to `None`, can't we derive default?

Reply:
Honestly, I didn't know that Default could be derived! Let me look into it and see. Expect another review request!